wmi_check/datadog_checks/wmi_check/wmi_check.py

```diff
@@ -37,7 +37,18 @@ def check(self, _):
         # Create or retrieve an existing WMISampler
         metric_name_and_type_by_property, properties = self.get_wmi_properties()
 
-        wmi_sampler = self.get_running_wmi_sampler(properties, self.filters, tag_by=self.tag_by)
+        tag_by_properties = ""
+        for t in self.tag_by.split(','):
+            t = t.strip()
+            if ' AS' in t or ' as' in t:
+                t_split = t.split(' AS') if ' AS' in t else t.split(' as')
+                t = t_split[0].strip()
+                tag_by_properties += t + ","
+            else:
+                tag_by_properties += t + ","
+        tag_by_properties = tag_by_properties.rstrip(',')
```

Comment:
Could you reuse `parse_alias` which also include Warning generation here, or there is more than meet the eye?

Reply:
Second part, as I have mention before, and it is not your code doing, although we are making it a bit worse, can we generally move config parsing into init function from check it subsequent functions, for performance reasons? Maybe in other PR?